pipeline/splicing/calcVarPriors.py

```diff
@@ -667,15 +686,305 @@ def getRefAltScores(refSeq, altSeq, donor=False):
+        subDonorBounds = refSpliceDonorBounds[exonNumber]
+        refSeq = getFastaSeq(varChrom, subDonorBounds["donorStart"], subDonorBounds["donorEnd"])
+        subMaxEntScanScore = runMaxEntScan(refSeq, donor=True)
+        subZScore = getZScore(subMaxEntScanScore, donor=True)
+        return {"maxEntScanScore": subMaxEntScanScore,
+                "zScore": subZScore}
+
+def isCIDomainInRegion(regionStart, regionEnd, boundaries, gene):
+    '''
+    Given a region of interest, boundaries (either enigma or priors) and gene of interest
+    Determines if there is an overlap between the region of interest and a CI domain
+    For minus strand gene (BRCA1) regionStart > regionEnd
+    For plus strand gene (BRCA2) regionStart < regionEnd
+    Returns True if there is an overlap, False otherwise
+    '''
+    if gene == "BRCA1":
+        for domain in brca1CIDomains[boundaries].keys():
+            domainStart = brca1CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca1CIDomains[boundaries][domain]["domEnd"]
+            overlap = range(max(regionEnd, domainEnd), min(regionStart, domainStart) + 1)
+            if len(overlap) > 0:
+                return True
+    elif gene == "BRCA2":
+        for domain in brca2CIDomains[boundaries].keys():
+            domainStart = brca2CIDomains[boundaries][domain]["domStart"]
+            domainEnd = brca2CIDomains[boundaries][domain]["domEnd"]
+            overlap = range(max(regionStart, domainStart), min(regionEnd, domainEnd) + 1)
+            if len(overlap) > 0:
+                return True
+    return False
+
+def getRefExonLength(variant):
+    '''Given a variant, returns the length of the reference exon'''
+    if varInExon(variant) == True:
+        varExonNum = getVarExonNumberSNS(variant)
+        exonBounds = getExonBoundaries(variant)
+        if getVarStrand(variant) == "-":
+            varExonStart = int(exonBounds[varExonNum]["exonStart"])
+            # varExonEnd - 1 to account for RefSeq numbering which starts to the right of the first base
+            varExonEnd = int(exonBounds[varExonNum]["exonEnd"]) - 1
+            exonLength = varExonStart - varExonEnd
+        else:
+            # varExonStart -1 to account for RefSeq numbering which starts to the right of the first base
+            varExonStart = int(exonBounds[varExonNum]["exonStart"]) - 1
+            varExonEnd = int(exonBounds[varExonNum]["exonEnd"])
+            exonLength = varExonEnd - varExonStart
+        return exonLength
+
+def getNewSplicePosition(varGenPos, varStrand, varWindowPos, inFirstThree):
+    '''
+    Given a variant's:
+    genetic postion, strand, sliding window position with max MES score AND
+      whether that position is within first three bases of highest scoring window
+    Returns the position where splicing occurs for a de novo splice donor
+    '''
+    if varStrand == "+":
+        if inFirstThree == False:
+            newSplicePos = int(varGenPos) - (varWindowPos - 3)
```

Comment:
Same feedback as before.  Rather than using the literal "3", use a parameter to indicate the width of the exonic portion of the splice site window.